### PR TITLE
housekeeping for tag url

### LINKED ISSUE #16641: 🧹 [UI House Keeping] `TagUrl`
### Description
Ensure that `TagUrl` adheres to all of the following conventions and standards

- [ ] Has a `className` prop and the PropType descriptions are all the same
- [ ] Prop table in MDX docs have the "Accepts all Box component props" description and link
- [ ] We are consistent when using the same prop names like `size` and are suggesting the use of the generalized `design-system.js` constants e.g. `SIZES` as the primary option but noting the component consts in the documentation and using them for propType validation and storybook controls only
- [ ] Standardize all similar prop names for images `imgSrc`, `imgAlt`(html element + attribute) (needs audit)
- [ ] We have a story for each component prop and we use the prop name verbatim e.g. `size` prop would be `export const Size = (args) => (`
- [ ] We have the accompanying documentation for each component prop and we use the prop name verbatim e.g. `size` prop would be `### Size`
- [ ] Are multiple props stories allowed? e.g. `Color, Background Color And Border Color` story in `base-avatar` - [ ] yes when it makes sense to
- [ ] All Base components follow the suffix convention e.g. `ButtonBase`
- [ ] All Base component MDX documentation have the base component notification at the top
- [ ] Add `mm-` prefix to all classNames
- [ ] className is kebab case version of the component name
- [ ] Spread base components props and reduce duplication of props when props aren't being changed and remain the same for both variant and base components
- [ ] Add component to root `index.js` file in component-library
- [ ] Add locals for any default text I18nContext as default context 
- [ ] Add any "to dos" with a `// TODO:` comment so we can search for them at a later date e.g. blocking components etc
- [ ] Add snapshot testing
- [ ] Add pixel values to propType descriptions if we use abstracted prop types that relate to pixel values e.g. `SIZE.MD (32px)`
- [ ] Each prop section in the MDX docs should have: a heading, a description, a story and an example code snipped

### PR DESCRIPTION
**Pull request by NidhiKJha**
* Fixes #16641 

### Description
This PR is to ensure that `TagUrl` adheres to all of the following conventions and standards

- [x] Has a `className` prop and the PropType descriptions are all the same
- [x] Prop table in MDX docs have the "Accepts all Box component props" description and link
- [ ] We are consistent when using the same prop names like `size` and are suggesting the use of the generalized `design-system.js` constants e.g. `SIZES` as the primary option but noting the component consts in the documentation and using them for propType validation and storybook controls only
- [x] Standardize all similar prop names for images `src`, `imgAlt`(html element + attribute) (needs audit)
- [x] We have a story for each component prop and we use the prop name verbatim e.g. `size` prop would be `export const Size = (args) => (`
- [x] We have the accompanying documentation for each component prop and we use the prop name verbatim e.g. `size` prop would be `### Size`
- [ ] Are multiple props stories allowed? e.g. `Color, Background Color And Border Color` story in `base-avatar` - [ ] yes when it makes sense to
- [ ] All Base components follow the suffix convention e.g. `ButtonBase`
- [x] All Base component MDX documentation have the base component notification at the top
- [x] Add `mm-` prefix to all classNames
- [x] className is kebab case version of the component name
- [x] Spread base components props and reduce duplication of props when props aren't being changed and remain the same for both variant and base components
- [x] Add component to root `index.js` file in component-library
- [x] Add locals for any default text I18nContext as default context 
- [x] Add any "to dos" with a `// TODO:` comment so we can search for them at a later date e.g. blocking components etc
- [x] Add snapshot testing
- [ ] Add pixel values to propType descriptions if we use abstracted prop types that relate to pixel values e.g. `SIZE.MD (32px)`
- [x] Each prop section in the MDX docs should have: a heading, a description, a story and an example code snipped

### CODE REVIEW

File: ui/components/component-library/tag-url/README.mdx [hunk 1]
```diff
@@ -16,55 +16,70 @@ The `TagUrl` accepts all props below as well as all [Box](/docs/ui-components-ui
...
 ```jsx
 
 import { TagUrl } from '../../ui/component-library/tag-url';
 
-  <TagUrl src="https://peepeth.com/favicon-32x32.png" />
-  <TagUrl src="https://1inch.exchange/assets/favicon/favicon-32x32.png"/>
-  <TagUrl src="https://uniswap.org/favicon.ico" />
+<TagUrl label="https://app.uniswap.org" actionButtonLabel="Permissions"/>
 
 ```
```
Comment (georgewrmarshall) [inline ui/components/component-library/tag-url/README.mdx:33]: nit: Formatting

```jsx
import { TagUrl } from '../../ui/component-library';

<TagUrl
  label="app.uniswap.org"
  actionButtonLabel="Permissions"
  src="https://uniswap.org/favicon.ico"
/>;
```

File: ui/components/component-library/tag-url/README.mdx [hunk 2]
```diff
@@ -16,55 +16,70 @@ The `TagUrl` accepts all props below as well as all [Box](/docs/ui-components-ui
...
 ```jsx
 
 import { TagUrl } from '../../ui/component-library/tag-url';
 
-  <TagUrl label="https://widget.getacute.io" />
-  <TagUrl label="app.uniswap.org" />
-  <TagUrl label="https://metamask.github.io" />
+<TagUrl label="https://app.uniswap.org" showLockIcon/>
 
 ```
```
Comment (georgewrmarshall) [inline ui/components/component-library/tag-url/README.mdx:49]: nit: formatting

```jsx
import { TagUrl } from '../../ui/component-library';

<TagUrl
  label="app.uniswap.org"
  src="https://uniswap.org/favicon.ico"
  showLockIcon
/>;

File: ui/components/component-library/tag-url/README.mdx [hunk 3]
```diff
@@ -16,55 +16,70 @@ The `TagUrl` accepts all props below as well as all [Box](/docs/ui-components-ui
...
+```jsx
 
-Use the `showLockIcon` prop to render a lock icon next to the `label`. It's intended use is to display if the url uses `https`. Props for the lock icon can be changed using the `lockIconProps`.
+import { TagUrl } from '../../ui/component-library/tag-url';
+
+  <TagUrl src="https://peepeth.com/favicon-32x32.png" />
+  <TagUrl src="https://1inch.exchange/assets/favicon/favicon-32x32.png"/>
+  <TagUrl src="https://uniswap.org/favicon.ico" />
+
+```
```
Comment (georgewrmarshall) [inline ui/components/component-library/tag-url/README.mdx:67]: Formatting and import statement. Not added a fall back component. Will note in stories file too

```jsx
import { TagUrl } from '../../ui/component-library';

<TagUrl
  label="peepeth.com"
  src="https://peepeth.com/favicon-32x32.png"
/>
<TagUrl
  label="1inch.exchange"
  src="https://1inch.exchange/assets/favicon/favicon-32x32.png"
/>
<TagUrl
  label="app.uniswap.org"
  src="https://uniswap.org/favicon.ico"
/>
<TagUrl label="fallback" src="" />
```

File: ui/components/component-library/tag-url/README.mdx [hunk 4]
```diff
@@ -16,55 +16,70 @@ The `TagUrl` accepts all props below as well as all [Box](/docs/ui-components-ui
...
 
+```jsx
+
+import { TagUrl } from '../../ui/component-library/tag-url';
+
+  <TagUrl label="https://peepeth.com" />
+  <TagUrl label="app.uniswap.org" />
+  <TagUrl label="https://metamask.github.io" />
+
+```
```
Comment (georgewrmarshall) [inline ui/components/component-library/tag-url/README.mdx:85]: nit: formatting

```jsx
import { TagUrl } from '../../ui/component-library';

<TagUrl label="https://peepeth.com/" />
<TagUrl label="app.uniswap.org" />
<TagUrl label="https://metamask.github.io" />
```

**Review by georgewrmarshall — COMMENTED** [review]
Looking good! Some small suggestions.
- remove `https://` from all the label examples in `.stories.js` and `.mdx` docs
- Not sure if we should show the actionButton and lock in the default story? I think we should lean towards showing all options so discoverability is high on the first story? WDYT?

Screenshot 2022-12-02 at 6 31 52 PM

**Review by georgewrmarshall — APPROVED** [review]
Looking good! I think I made a mistake in my review and suggested the wrong story update. I've taken the liberty of fixing it to save some time https://github.com/MetaMask/metamask-extension/pull/16723/commits/7f6ba2eceea2f5bdacad1d4655ef9ab3c8ee8cc4 and from me making the mistake again. My baad 🙇  Not sure if you're rebasing or merging from develop. So haven't fixed the failing build. Updating the branch should fix 🤔

**Review by garrettbear — APPROVED** [review]

**Pull request merged**